src/displayapp/screens/Gallery.cpp

```diff
@@ -0,0 +1,105 @@
+#include <nrf_log.h>
+#include "displayapp/screens/Gallery.h"
+#include "displayapp/DisplayApp.h"
+
+using namespace Pinetime::Applications::Screens;
+
+Gallery::Gallery(DisplayApp* app, Pinetime::Controllers::FS& filesystem) : Screen(app), filesystem(filesystem) {
+  ListDir();
+
+  if (nScreens == 0) {
+    lv_obj_t* title = lv_label_create(lv_scr_act(), nullptr);
+    lv_label_set_text_static(title, "no images found");
+    lv_label_set_align(title, LV_LABEL_ALIGN_CENTER);
+    lv_obj_align(title, lv_scr_act(), LV_ALIGN_CENTER, 0, 0);
+  } else {
+    Open(0, DisplayApp::FullRefreshDirections::None);
+  }
+}
+
+Gallery::~Gallery() {
+  lv_obj_clean(lv_scr_act());
+}
+
+bool Gallery::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
+  switch (event) {
+    case Pinetime::Applications::TouchEvents::SwipeRight:
+      return Open(index - 1, DisplayApp::FullRefreshDirections::Right);
+    case Pinetime::Applications::TouchEvents::SwipeLeft:
+      return Open(index + 1, DisplayApp::FullRefreshDirections::Left);
+    case Pinetime::Applications::TouchEvents::LongTap:
+    case Pinetime::Applications::TouchEvents::DoubleTap:
+      current->ToggleInfo();
+      return true;
+  }
+  return false;
+}
+
+void Gallery::ListDir() {
+  lfs_dir_t dir = {0};
+  lfs_info info = {0};
+  nScreens = 0;
+
+  int res = filesystem.DirOpen(directory, &dir);
+  if (res != 0) {
+    NRF_LOG_INFO("[Gallery] can't find directory");
+    return;
+  }
+  while (filesystem.DirRead(&dir, &info)) {
+    if (info.type == LFS_TYPE_DIR)
+      continue;
+    nScreens++;
+  }
+  assert(filesystem.DirClose(&dir) == 0);
```

Comment:
I investigated the issue a bit further and have the feeling that the issue might be related to LittleFS. I have commented out almost all functionality and am just running this `DirRead` loop. Everything else, including the creation of the actual screens in `Gallery::Open`, is commented out. This still crashes on my watch when I do the following
 * open the gallery app
 * ignore the fact that it's completely empty
 * exit the gallery app,
 * go to any setting (eg. Time format)
 * try to change it and exit setting

To confirm it's related to the `DirRead` and nothing else, I then hardcoded a file path in `Gallery::Open` and disable the `DirRead` completely. In that case the bug did not appear. So I'd guess it's related to this loop

Reply:
I think i just found the issue. You are calling DirClose inside the assert which you should not do. If your move the call to a seperate variable then the issue seems to disappear. My watch ran pretty stable after i changed it

Reply:
Very nice, thanks for figuring it out :partying_face:  I've been using a modified version of the PR for a while now that circumvented the `DirRead` problem but was quite annoying to use. I'll fix this asap

Reply:
fixed this in 15bce7f5